src/gps_to_pose_conversion_node.cpp

```diff
@@ -76,6 +77,13 @@ void gps_callback(const sensor_msgs::NavSatFixConstPtr& msg)
   pose_msg->pose.pose.position.z = z;
   pose_msg->pose.pose.orientation = g_latest_imu_msg.orientation;
 
+  // Fill up position message
+  geometry_msgs::PointStampedPtr position_msg(
+    new geometry_msgs::PointStamped);
+  position_msg->header = pose_msg->header;
+  position_msg->header.frame_id = "world";
+  position_msg->point = pose_msg->pose.pose.position;
+
   if (g_got_pressure) {
```

Comment:
I've added a subscriber for "external_altitude" input, replacing the one we had originally for pressure only. In this code, the altitude is published in the GPS position message, provided the node receives messages on the topic. Basically, if a remapping to "external_altitude" exists, the node will use this altitude; if not, it will take the GPS altitude. I think a Boolean parameter would be a bit redundant in this case, as we can simply switch between the two by controlling the topic remapping. What do you think, what would be the more elegant way of doing this?